apps/backend/src/experiments/application/use-cases/experiment-members/add-experiment-members.ts

```diff
@@ -0,0 +1,103 @@
+import { Injectable, Logger } from "@nestjs/common";
+
+import {
+  Result,
+  success,
+  failure,
+  AppError,
+} from "../../../../common/utils/fp-utils";
+import {
+  ExperimentMemberDto,
+  ExperimentMemberRole,
+} from "../../../core/models/experiment-members.model";
+import { ExperimentDto } from "../../../core/models/experiment.model";
+import { ExperimentMemberRepository } from "../../../core/repositories/experiment-member.repository";
+import { ExperimentRepository } from "../../../core/repositories/experiment.repository";
+
+@Injectable()
+export class AddExperimentMembersUseCase {
+  private readonly logger = new Logger(AddExperimentMembersUseCase.name);
+
+  constructor(
+    private readonly experimentRepository: ExperimentRepository,
+    private readonly experimentMemberRepository: ExperimentMemberRepository,
+  ) {}
+
+  async execute(
+    experimentId: string,
+    members: { userId: string; role?: ExperimentMemberRole }[],
+    currentUserId: string,
+  ): Promise<Result<ExperimentMemberDto[]>> {
+    this.logger.log(
+      `Adding members [${members.map((m) => m.userId).join(", ")}] to experiment ${experimentId} by user ${currentUserId}`,
+    );
+
+    // Check if the experiment exists
+    const experimentResult =
```

Comment:
Think this could be skipping i am guessing the get members will throw if the experiment does not exits, also it might be a good option to add a utility method on the repository level like isAdmin that takes the user ID and experiment ID and returns boolean